### ISSUE TITLE
Always empty intersection when building the train dictionary

### ISSUE THREAD

**Issue by lipingtang17** [muyeby/BiAAE#1]
Dear Xuefeng,

Thanks for your sharing.
But when I use your code to train the model, I always get empty intersection when building the train dictionary:
I trained the model by "bash run.sh", set the DATA_DIR as my own direction of embedding files and set the training epoch as 50. I use the default command in your "run.sh": 
`python main-trainer.py --seed $s --dataset wiki --num_epochs $N --src_lang $src --src_emb $DATA_DIR/wiki.$src.vec --tgt_lang $tgt --tgt_emb $DATA_DIR/wiki.$tgt.vec --norm_embeddings $norm --cuda_device $dev --exp_id $dir --state 3 --gate $gate`
The following is part of the results:
```
Epoch 47 : Discriminator Loss: 0.711, Discriminator Accuracy: 0.995, Generator Loss: 5.026, Time elapsed 1.11 mins
csls_knn_10 takes 66.86s
Method:csls_knn_10 score:75.9333
Building the train dictionary ...
Empty intersection ...
Epoch 48 : Discriminator Loss: 0.709, Discriminator Accuracy: 0.995, Generator Loss: 5.106, Time elapsed 1.18 mins
csls_knn_10 takes 66.64s
Method:csls_knn_10 score:69.6667
Building the train dictionary ...
Empty intersection ...
Epoch 49 : Discriminator Loss: 0.709, Discriminator Accuracy: 0.995, Generator Loss: 5.050, Time elapsed 1.13 mins
csls_knn_10 takes 67.02s
Method:csls_knn_10 score:73.6000
Building the train dictionary ...
Empty intersection ...
```

**Comment by goodbai-nlp** [comment]
Hi, Liping

Sorry for my late response. It is reasonable to get an "Empty intersection" sometimes since the model takes "S2T&T2S"  as a dictionary building method by default. https://github.com/muyeby/BiAAE/blob/436b702a7672da8abeeacfe68f436dd2e121439a/src/main-trainer.py#L85
You can 

1. change the dictionary building method to "S2T" (this will change the model selection criterion and performance).
2. keep the default parameter and ignore this (if the model performance is ok).
